### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by pvenkatakrishnan**
* Fixing issues: https://github.com/krakenjs/grunt-localizr/issues/9 https://github.com/krakenjs/grunt-localizr/issues/7
* cleanup,
* moving to bluebird
* increasing coverage

### CODE REVIEW

**Comment by aredridel** [comment]
:+1: Utterly solid.

File: lib/localeList.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                    result.push(entry.country);
+                }
+                result = result.concat(entry.langs.map(function(lang) {
+                    return path.join(entry.country, lang);
+                }));
+            });
+            cb(null, result);
+        });
+};
+
```
Comment (grawk) [inline lib/localeList.js]: I feel like this function needs a jsdoc comment block

Comment (pvenkatakrishnan) [inline lib/localeList.js]: will add more comments

File: lib/localeList.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+var fs = require('fs');
+var path = require('path');
+var BB = require('bluebird');
+
+var readdir = BB.promisify(fs.readdir);
+
+module.exports = function getLocaleList(root, cb) {
```
Comment (grawk) [inline lib/localeList.js]: Some comments in this function would be helpful. Maybe just a jsdoc style inputs/outputs/brief description block

File: lib/localeList.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ *
+ * @param root: The content bundle root, for eg: path/to/locales/
+ * @param cb: The callback to invoke when the locale list is ready
+ * The callback is node style: takes two args (err, result) where:
+ *      err: Any error while generating the locale list in which case the result is undefined.
+        result: An array of locale paths eg: [ 'US', 'US/en', 'US/es', 'US/fr' ]
+ */
+module.exports = function getLocaleList(root, cb) {
+    readdir(root).then(function (countries) {
```
Comment (grawk) [inline lib/localeList.js]: Do we care that passing a non-existent path causes an unhandled exception?

File: test/index.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ var grunt = require('grunt'),
 test('Grunt-localizr', function (t) {
```
Comment (grawk) [inline test/index.js]: We _could_ create a separate test file `test/localeList.js` which isolates localeList. I've done that in my local repo to work through my code review. It could help us to more easily test any other corner case assumptions.

Comment (pvenkatakrishnan) [inline test/index.js]: could you pr those in then ? we can merge mine and then yours as well against that.

Comment (grawk) [inline test/index.js]: I've submitted a PR to your master branch: https://github.com/pvenkatakrishnan/grunt-localizr/pull/1

**Comment by grawk** [comment]
:+1:
